_posts/2025-04-16-tha-various-tiers-of-tiered-storage.md

```diff
@@ -0,0 +1,454 @@
+
+The performance profile of the remote storage is usually also different from the block storage used for the local tier.
+In particular the latency is often higher.
+So accessing your older data that were offloaded to the remote tier might not be as fast as accessing the data kept locally.
+
+Luckily, Apache Kafka allows us to configure the tiered storage on a per-topic basis.
+So unlike other streaming platforms that rely only on the remote storage tier, you can decide when will tiered storage be used.
+That way, you can keep the topics where low latency is critical or where repeated consumption would make the costs skyrocket as local only.
+
+#### What makes a good remote storage?
+
+The remote storage is used by all Kafka brokers from a given cluster.
+The data stored in the remote tier are not assigned to any particular broker.
+The data for given partition will be always written to the remote storage by the broker hosting the leader replica.
+And they will be read by the brokers hosting the leader or follower replicas.
+But when partition replicas are reassigned between the brokers, their data will remain unchanged in the remote storage tier.
+So whatever type of storage you use as the remote tier, it must allow shared read and write access to all the data from all brokers.
+
+Another important aspect when choosing the right remote storage is its performance, scalability, durability, and availability.
+The actual requirements depend on your use-cases or on the type of environment where you use it.
+But your Kafka cluster will be only as good as its weakest part.
+So if you require high availability, durability, and performance, you have to make sure that your remote storage supports these requirements.
+
+There are many technologies that might match these requirements.
+Apart from the object storage, it can be for example HDFS storage or shared file storage.
+
+NOTE: Apache Kafka uses plugins for the different implementations of the remote storage tier.
+But no actual implementations are shipped as part of Apache Kafka itself.
+So unless you want to write your own plugin, you have to also make sure that there is already some plugin implemented by someone else and has the required quality.
+
+### Shared storage as a tiered storage
+
+While running high-quality object storage on-premise might be challenging, shared file storage is often already available.
+It might have for example the form of Network File System (NFS) volumes.
+While the quality of it might differ between the users and their environments, it is in many cases already used by other applications and has the required availability and reliability.
+
+NFS storage matches the requirements we listed in the previous section.
+You can use it as a shared read-write-many volume between all your Kafka brokers.
+And it can provide similar benefits to object storage.
```

Comment:
Do you think we can also mention the "security/privacy" in NFS, because the data is owned by yourselves, not by the the cloud provider?